src/main/java/io/confluent/connect/jdbc/source/TimestampIncrementingTableQuerier.java

```diff
@@ -143,6 +153,57 @@ protected void createPreparedStatement(Connection db) throws SQLException {
     stmt = dialect.createPreparedStatement(db, queryString);
   }
 
+  @Override
+  public void maybeStartQuery(Connection db) throws SQLException, ConnectException {
+    if (resultSet == null) {
+      this.db = db;
+      stmt = getOrCreatePreparedStatement(db);
+      resultSet = executeQuery();
+      String schemaName = tableId != null ? tableId.tableName() : null; // backwards compatible
+      ResultSetMetaData metadata = resultSet.getMetaData();
+      testForUnsupportedDatabaseColumnTypes(metadata);
+      schemaMapping = SchemaMapping.create(schemaName, metadata, dialect);
+    }
+  }
+
+  /**
+   * Do runtime checks on table metadata
+   * If SqlServer dialect (Sql Server 2016 or older), time stamp mode configured,
+   * and time stamp column is datetime, kill task
+   * @param metadata table meta data
+   * @throws ConnectException if run time check failed
+   */
+  private void testForUnsupportedDatabaseColumnTypes(
```

Comment:
This method should be in the dialect classes.

Reply:
+1

Reply:
I was going to do it but I held back because I thought moving this over to the Dialect would obscure the Connect Exception that would get thrown (because it would not be in the "main thread" of execution if that makes sense). 
Would you agree with me if I leave this function here but move it's logic into the Dialect class (this function would use a bool to determine if it needs to throw a Connect Exception)?